test/get-blur-areas.js

```diff
@@ -0,0 +1,320 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+
+describe('Connection with server part', () => {
+  // we show errors to user by alert.
+  // disable it for testing
+  alert = function() {
+  };
+
+  /**
+   * Helper function to check if the responses with rects are equal.
+   * These responses are objects, and simple '==' will
+   * compare references of objects, so it doesn't work
+   * @param {Array<Rect>} response1
+   * @param {Array<Rect>} response2
+   * @return {boolean}
+   */
+  function areRectResponsesEqual(response1, response2) {
+    if (!Array.isArray(response1) || !Array.isArray(response2)) {
+      return false;
+    }
+
+    if (response1.length !== response2.length) {
+      return false;
+    }
+
+    for (let i = 0; i < response1.length; i++) {
+      if (!areRectsEqual(response1[i], response2[i])) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /**
+   * Helper function to check if rect objects are equal.
+   * These rects are objects, and simple '==' will
+   * compare references of objects, so it doesn't work
+   * @param {Rect} rect1
+   * @param {Rect} rect2
+   * @return {boolean}
+   */
+  function areRectsEqual(rect1, rect2) {
+    if (!rect1.hasOwnProperty('leftX') || !rect1.hasOwnProperty('topY') ||
+        !rect1.hasOwnProperty('width') || !rect1.hasOwnProperty('height')) {
+      return false;
+    }
+
+    if (!rect2.hasOwnProperty('leftX') || !rect2.hasOwnProperty('topY') ||
+        !rect2.hasOwnProperty('width') || !rect2.hasOwnProperty('height')) {
+      return false;
+    }
```

Comment:
Can you make it as a loop over rect1 and rect2 and then over all these properties?